tests/print_ver.c

```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016 Red Hat, Inc.
+ * Copyright 2018 Red Hat, Inc.
```

Comment:
In the age of git, I suppose solely newest date makes the most sense,
it provides a basic estimate and if anyone needs exact legal dispute,
git provides exact data easily.

Reply:
I should have said "worst case estimate", which makes whole a lot tof sense,
at least to me.

Reply:
> In the age of git, I suppose solely newest date makes the most sense,
> it provides a basic estimate and if anyone needs exact legal dispute,
> git provides exact data easily.

Certainly, the git history provides a good claim to priority, but the copyright notice here is not valid.

See the U.S. Copyright Office's "Compendium of U.S. Copyright Office Practices" [ https://www.copyright.gov/comp3/ ], particularly "Chapter 2200: Notice of Copyright", sections 2205.1(A)  and 2205.1(F), or for a much more readable summary:

https://techwhirl.com/updating-copyright-notices/

The original date of publication must be in the notice for it to be valid. Giving a year range indicates that portions of the work were added in different years, though it cannot say which portions are which -- that is where the git history comes in handy.

Reply:
See the footer on redhat.com main page, then (*sigh*)

Reply:
Looking deeper in your linked article:

> What happens when the manual is revised? Interestingly, the
> Copyright Act doesn’t directly say. But the commonly accepted
> practice is to include multiple years in the copyright notice,
> indicating the various years in which various material in the
> overall work was first published. 

It may make sense for the material without rigorously tracked
history, but again, it's not the case with VCS-tracked project
(unless it's history got unclear on the initial import, which
is, again, not the case with libqb).

Furthermore, per
https://www.copyright.gov/help/faq/faq-definitions.html#notice:

> While use of a copyright notice was once required as a condition
> of copyright protection, it is now optional. 

So I see no contradiction when using mere single date (newest), making
it an estimate (version control will always have the last say) when
the given file at the current form goes out of copyright protection
(country-dependent as mentioned) and putting negligible (because of
VCS)onus on figuring out the exact data about its parts to anyone
wanting to use them beyond copyright-granted protection packed with
LGPL-granted provisions (that should be enough in most cases anyway).
Fair deal, isn't it?

Reply:
Purely as a matter of style I prefer the date "from-to" format as it gives some idea how long the file has been there without having to dig into the VCS history. Also, when things have been moved from one repo to another the dates can get unreliable. I'm not saying that's happened here, but if you keep the dates inside the file then you can be sure that it doesn't get lost ... unless someone removes it ;-)

Reply:
The legal form of the copyright notice is intentionally ambiguous to allow variety, but one point it is always clear on is that the original date of publication must be listed. The initial part may be "Copyright", "Copr.", or the copyright symbol, and the year may be in just about any recognizable form (numerals, spelled out, Roman numerals, etc.), but it must specify the year of initial publication. The git history is good for determining which portions of the work were created in which years, but the copyright notice should list the original year, the original year plus later years of modification, or a range from the original year to the year of latest modification.